beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestIntegrationVoidInvoice.java

```diff
@@ -129,4 +138,85 @@ public void testVoidInvoice() throws Exception {
+        assertNotNull(account);
+
+        add_AUTO_PAY_OFF_Tag(account.getId(), ObjectType.ACCOUNT);
+
+        final PlanPhaseSpecifier spec = new PlanPhaseSpecifier("pistol-monthly-notrial");
+
+        busHandler.pushExpectedEvents(NextEvent.INVOICE);
+        final InvoiceItem inputCredit = new CreditAdjInvoiceItem(null, account.getId(), startDate, "credit invoice", new BigDecimal("20.00"), account.getCurrency(), null);
+        invoiceUserApi.insertCredits(account.getId(), startDate, ImmutableList.of(inputCredit), true, null, callContext);
+        assertListenerStatus();
+
+        final BigDecimal accountBalance1 = invoiceUserApi.getAccountBalance(account.getId(), callContext);
+        final BigDecimal accountCBA1 = invoiceUserApi.getAccountCBA(account.getId(), callContext);
+
+        busHandler.pushExpectedEvents(NextEvent.BLOCK, NextEvent.CREATE, NextEvent.INVOICE);
+        final UUID entitlementId = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec, null, null, null), null, startDate, startDate, false, false, ImmutableList.<PluginProperty>of(), callContext);
+        final Entitlement bpEntitlement = entitlementApi.getEntitlementForId(entitlementId, callContext);
+        assertListenerStatus();
+
+        final Invoice invoice2 = invoiceChecker.checkInvoice(account.getId(), 2, callContext,
+                                                             new ExpectedInvoiceItemCheck(new LocalDate(2013, 6, 15), new LocalDate(2013, 7, 15), InvoiceItemType.RECURRING, new BigDecimal("19.95")),
+                                                             new ExpectedInvoiceItemCheck(new LocalDate(2013, 6, 15), new LocalDate(2013, 6, 15), InvoiceItemType.CBA_ADJ, new BigDecimal("-19.95")));
+
+
+        // 2013-07-01
+        clock.addDays(16);
+        busHandler.pushExpectedEvents(NextEvent.BLOCK, NextEvent.CANCEL, NextEvent.INVOICE);
+        bpEntitlement.cancelEntitlementWithPolicyOverrideBillingPolicy(EntitlementActionPolicy.IMMEDIATE, BillingActionPolicy.IMMEDIATE, ImmutableList.<PluginProperty>of(), callContext);
+        assertListenerStatus();
+
+        final Invoice invoice3 = invoiceChecker.checkInvoice(account.getId(), 3, callContext,
+                                                             new ExpectedInvoiceItemCheck(new LocalDate(2013, 7, 1), new LocalDate(2013, 7, 15), InvoiceItemType.REPAIR_ADJ, new BigDecimal("-9.31")),
+                                                             new ExpectedInvoiceItemCheck(new LocalDate(2013, 7, 1), new LocalDate(2013, 7, 1), InvoiceItemType.CBA_ADJ, new BigDecimal("9.31")));
+        // We disallow to void the invoice as it was repaired
+        try {
+            invoiceUserApi.voidInvoice(invoice2.getId(), callContext);
+            Assert.fail("Should fail to void a repaired invoice");
+        } catch (final RuntimeException e) {
+            if (!e.getMessage().contains("because it contains items being repaired")) {
```

Comment:
Nit: could do a proper assertion here (`assertTrue`).

Reply:
Fixed in https://github.com/killbill/killbill/pull/1499/commits/3537a156a019ec744647e49609a35a8774977e3d